HPXMLtoOpenStudio/resources/EPvalidator.xml

```diff
@@ -568,23 +607,35 @@
       <sch:assert role='ERROR' test='count(h:HeatingCapacity) &lt;= 1'>Expected 0 or 1 element(s) for xpath: HeatingCapacity</sch:assert>
       <sch:assert role='ERROR' test='count(h:CoolingCapacity) &lt;= 1'>Expected 0 or 1 element(s) for xpath: CoolingCapacity</sch:assert>
       <sch:assert role='ERROR' test='count(h:CoolingSensibleHeatFraction) &lt;= 1'>Expected 0 or 1 element(s) for xpath: CoolingSensibleHeatFraction</sch:assert>
-      <sch:assert role='ERROR' test='not(h:BackupSystemFuel) or count(h:BackupSystemFuel[text()="electricity" or text()="natural gas" or text()="fuel oil" or text()="fuel oil 1" or text()="fuel oil 2" or text()="fuel oil 4" or text()="fuel oil 5/6" or text()="diesel" or text()="propane" or text()="kerosene" or text()="wood" or text()="wood pellets"]) = 1'>Expected 1 element(s) for xpath: [not(BackupSystemFuel)] | BackupSystemFuel[text()="electricity" or text()="natural gas" or text()="fuel oil" or text()="fuel oil 1" or text()="fuel oil 2" or text()="fuel oil 4" or text()="fuel oil 5/6" or text()="diesel" or text()="propane" or text()="kerosene" or text()="wood" or text()="wood pellets"]</sch:assert> <!-- See [HeatPumpBackup] -->
+      <sch:assert role='ERROR' test='count(h:BackupSystemFuel) &lt;= 1'>Expected 0 or 1 element(s) for xpath: BackupSystemFuel</sch:assert> <!-- See [HeatPumpBackup] and [HeatPumpBackupSystemFuel]-->
```

Comment:
This is good. But it might appear strange to a developer/user why BackupSystemFuel is separated into nested rules but not HeatPumpFuel. I'm not really sure how to handle it. We _need_ nested rules if the element is optional, whereas nested rules _could_ (but don't have to be) used if the element is required. I guess my worry is that a developer will not realize this subtlety and will create an optional element w/o nested rules in the future, which would be incorrect.

So should we move in the direction of nesting rules as much as possible, regardless of whether the element is optional or not? I'm not really sure.

Reply:
Hmm. As you mention, there's a high chance that a developer will not realize this subtlety. I think we should use nesting rules as much as possible. It will result in some additional lines for assert tests but that way, the code will be consistent.